casper/src/main/scala/coop/rchain/casper/MultiParentCasperImpl.scala

```diff
@@ -196,6 +200,7 @@ class MultiParentCasperImpl[F[_]: Sync: Concurrent: Log: Time: SafetyOracle: Las
                     case false =>
                       ().pure[F]
                   }
+      _ <- blocksInProcessing.update(_ - b.blockHash)
```

Comment:
If block is already in processing and if it's called the second time, it cannot acquire lock so it will be deleted from the processing set. This doesn't seems right.

Reply:
Yes you are right, thanks